nestjs/src/courses/team-distribution/dto/team-distribution.dto.ts

```diff
@@ -54,3 +56,50 @@ export class TeamDistributionDto {
   @ApiProperty()
   public minTotalScore: number;
 }
+
+@ApiResponse({})
+export class TeamDistributionDetailedDto {
+  constructor(distribution: TeamDistribution, team?: TeamDto) {
+    this.studentsWithoutTeamCount = distribution.studentsWithoutTeam.length;
+    this.teamsCount = distribution.teams.length;
+    this.id = distribution.id;
+    this.name = distribution.name;
+    this.myTeam = team;
+    this.minStudents = distribution.minStudents;
+    this.maxStudents = distribution.maxStudents;
+    this.studentsCount = distribution.studentsCount;
+    this.strictStudentsCount = distribution.strictStudentsCount;
+    this.courseId = distribution.courseId;
+  }
+
+  @ApiProperty()
+  public id: number;
+
+  @ApiProperty()
+  public courseId: number;
+
+  @ApiProperty()
+  public name: string;
+
+  @ApiProperty()
+  public studentsWithoutTeamCount: number;
+
+  @ApiProperty()
+  public teamsCount: number;
+
+  @ApiProperty()
+  @IsOptional()
+  public myTeam?: TeamDto;
+
+  @ApiProperty()
+  public minStudents: number;
+
+  @ApiProperty()
+  public maxStudents: number;
+
+  @ApiProperty()
+  public studentsCount: number;
+
+  @ApiProperty()
+  public strictStudentsCount: boolean;
```

Comment:
what does `strict` in this context mean? consider adding comment

Reply:
fixed